File: CS50’s Introduction to Computer Science/Week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int L = count_letters(text), W = count_words(text), S = count_sentences(text);

    float letters = (float) L / (float) W * 100;

    float sentences = (float) S / (float) W * 100;

    int index = round(0.0588 * letters - 0.296 * sentences - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 16 && index >= 1)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

int count_letters(string text)
{
    int numle = 0;
    int value = strlen(text);
    for (int i = 0; i < value; i++)
    {
        if (isalpha(text[i]))
        {
            numle++;
        }
    }
    return numle;
}

int count_words(string text)
{
    int numwo = 1;
    int value = strlen(text);
    for (int i = 0; i < value; i++)
    {
        if (isspace(text[i]))
        {
            numwo++;
        }
    }
    return numwo;
}

int count_sentences(string text)
{
    int numsen = 0;
    int value = strlen(text);
    for (int i = 0; i < value; i++)
    {
        if ('!' == text[i] || '.' == text[i] || '?' == text[i])
        {
            numsen++;
        }
    }
    return numsen;
}

File: CS50’s Introduction to Computer Science/Week 6/sentimental/readability.py
from cs50 import get_string


def count_letters(text):
    numle = 0
    for char in text:
        if char.isalpha():
            numle += 1
    return numle


def count_words(text):
    numwo = 1
    for char in text:
        if char.isspace():
            numwo += 1
    return numwo


def count_sentences(text):
    numsen = 0
    for char in text:
        if char in ['!', '.', '?']:
            numsen += 1
    return numsen


text = get_string("Text: ")

l = count_letters(text)

w = count_words(text)

s = count_sentences(text)

letters = l/w * 100

sentences = s/w * 100

index = round(0.0588 * letters - 0.296 * sentences - 15.8)

if index > 16:
    print("Grade 16+")
elif 1 <= index <= 16:
    print(f"Grade {index}")
else:
    print("Before Grade 1")
